datajs/src/file-base.js

```diff
@@ -140,13 +142,36 @@ export class FileInterface extends File {
     return this.descriptor.name
   }
 
-  async hash() {
-    const text = await this.descriptor.text()
-    return crypto.createHash('md5').update(text).digest('hex')
-  }
-
-  async hashSha256() {
-    const text = await this.descriptor.text()
-    return crypto.createHash('sha256').update(text).digest('hex')
+  generateHash(hashType, cbProgress) {
+    return new Promise((resolve, reject) => {
+      let newHash =  hashType === "md5" ? CryptoJS.algo.MD5.create() : CryptoJS.algo.SHA256.create();
+      readChunked(this.descriptor, (chunk, offs, total) => {
+        newHash.update(CryptoJS.enc.Latin1.parse(chunk));
+        if (cbProgress) {
+          cbProgress(offs / total);
+        }
+      }, err => {
```

Comment:
@mariorodeghiero why do you use `CryptoJS` instead of `crypto`? What was the output difference? Those two modules should return the same result. If you don't get the same, then something is wrong with the way we use it. I would recommend using only `crypto`, becuase `CryptoJS` is nothing else than replicating the same funcitons in the `crypto`

Reply:
because of time, I just push the old code to be able to generate for a large files. Anyway, the new code is here.

```
async generateHash(hashType, cbProgress) {
    return new Promise((resolve, reject) => {
      let newHash =  hashType === "md5" ? crypto.createHash('md5') : crypto.createHash('sha256');
      readChunked(this.descriptor, (chunk, offs, total) => {
        newHash.update(chunk);
        if (cbProgress) {
          cbProgress(offs / total);
        }
      }, err => {
        if (err) {
          reject(err);
        } else {
          // TODO: Handle errors
          let hashHex = newHash.digest('hex')
          resolve(hashHex);
        }
      });
    });
  }
```